### PR TITLE
COMP: Bump CastXML amd64 binaries to 0.4.5 for Windows 11 support

### PR DESCRIPTION
**Pull request by thewtex**

### CODE REVIEW

**Comment by thewtex** [comment]
Closes #3216

**Review by bradking — APPROVED** [review]

**Review by dzenanz — APPROVED** [review]

**Comment by thewtex** [comment]
\azp run ITK.macOS.Python

**Comment by aylward** [comment]
Looks like the Mac version of castxml has a packaging error:
https://dev.azure.com/itkrobotmacospython/ITK.macOS.Python/_build/results?buildId=6827&view=logs&j=a5e52b91-c83f-5429-4a68-c246fc63a4f7&t=5add3200-7efa-59c0-0a45-43bce0c77c69&l=5424

**Comment by aylward** [comment]
Did not work on Windows 11.

> ninja

FAILED: Wrapping/itkVariableSizeMatrix.xml
cmd.exe /C "cd /D C:\src\ITK-Release\Wrapping\Modules\ITKCommon && C:\src\ITK-Release\Wrapping\Generators\CastXML\castxml\bin\castxml.exe -o C:/src/ITK-Release/Wrapping/itkVariableSizeMatrix.xml --castxml-gccxml --castxml-start _wrapping_ --castxml-cc-msvc ( "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe" /DWIN32 /D_WINDOWS /GR /EHsc -W3 /bigobj /bigobj -std:c++14 ) -fexceptions -w -c @C:/src/ITK-Release/Wrapping/ITKCommon.castxml.inc C:/src/ITK-Release/Wrapping/itkVariableSizeMatrix.cxx"
'C:\src\ITK-Release\Wrapping\Generators\CastXML\castxml\bin\castxml.exe' is not recognized as an internal or external command, operable program or batch file.

> ls /c/src/ITK-Release/Wrapping/Generators/CastXML/castxml/bin

castxml

> cd /c/src/ITK-Release/Wrapping/Generators/CastXML/castxml/bin
> mv castxml castxml.exe
> cd /c/src/ITK-Release
> ninja

FAILED: Wrapping/ITKCommonBase.xml
cmd.exe /C "cd /D C:\src\ITK-Release\Wrapping\Modules\ITKCommon && C:\src\ITK-Release\Wrapping\Generators\CastXML\castxml\bin\castxml.exe -o C:/src/ITK-Release/Wrapping/ITKCommonBase.xml --castxml-gccxml --castxml-start _wrapping_ --castxml-cc-msvc ( "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe" /DWIN32 /D_WINDOWS /GR /EHsc -W3 /bigobj /bigobj -std:c++14 ) -fexceptions -w -c @C:/src/ITK-Release/Wrapping/ITKCommon.castxml.inc C:/src/ITK-Release/Wrapping/ITKCommonBase.cxx"
This version of C:\src\ITK-Release\Wrapping\Generators\CastXML\castxml\bin\castxml.exe is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher.

File: Wrapping/Generators/CastXML/CMakeLists.txt [hunk 1]
```diff
@@ -37,19 +37,19 @@ else()
...
-    set(_castxml_hash 54a10b68399d17d41360a28c2f5091ed2dfcd104b8acb0f68a6224263de56348792af453355c1b243fb019efe476d01ad68dbd68db0908ce9cf62f606a5a31bb)
+    set(_castxml_hash 5faef93541cb5f30414286a844e47b5732777f26246c6e0933170668e9cf2ff4f3fa4292e861e73e91ed50971caa9d6e7481eead9c9cefda6d54a9042634bf5c)
     set(_castxml_url "https://data.kitware.com/api/v1/file/hashsum/sha512/${_castxml_hash}/download")
     set(_download_castxml_binaries 1)
 
   # If 64 bit Mac OS X build host ( >= 10.9, Mavericks), use the CastXML binary
   elseif(CMAKE_HOST_SYSTEM_NAME MATCHES "Darwin" AND CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64" AND (NOT CMAKE_HOST_SYSTEM_VERSION VERSION_LESS "13.0.0"))
 
-    set(_castxml_hash f7d62660b8c3f19481c0f2b87b144f9cd2a500ad94b1b50b27c1f36d4c0096ca46df9c8a833426392a1d5152bd58dabd2bb0b525f56e341f5310afcf98013748)
+    set(_castxml_hash 56cf92eb9ca543a6178689643d32a455da1f92da78ca58a8cb671798ac9f903dc6b687be1d3824ab0211ed49724b62185a21874810139de7dbb6cb39f63b860f)
```
Comment (aylward) [inline Wrapping/Generators/CastXML/CMakeLists.txt:47]: This is the hash for the windows version, not the mac version.

https://data.kitware.com/#item/622961504acac99f42134aab

File: Wrapping/Generators/CastXML/CMakeLists.txt [hunk 2]
```diff
@@ -37,19 +37,19 @@ else()
   # If 64 bit Windows build host, use the CastXML binary
   elseif(CMAKE_HOST_SYSTEM_NAME MATCHES "Windows" AND CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "AMD64")
 
-    set(_castxml_hash 54a10b68399d17d41360a28c2f5091ed2dfcd104b8acb0f68a6224263de56348792af453355c1b243fb019efe476d01ad68dbd68db0908ce9cf62f606a5a31bb)
+    set(_castxml_hash 5faef93541cb5f30414286a844e47b5732777f26246c6e0933170668e9cf2ff4f3fa4292e861e73e91ed50971caa9d6e7481eead9c9cefda6d54a9042634bf5c)
```
Comment (aylward) [inline Wrapping/Generators/CastXML/CMakeLists.txt:40]: The proper hash is 
56cf92eb9ca543a6178689643d32a455da1f92da78ca58a8cb671798ac9f903dc6b687be1d3824ab0211ed49724b62185a21874810139de7dbb6cb39f63b860f
See
https://data.kitware.com/#item/622961504acac99f42134aab

**Review by aylward — CHANGES_REQUESTED** [review]
Hash tags got swapped.   Not certain what is the right version for Darwin x86_64 - didn't see it on data.kitware.com

File: Wrapping/Generators/CastXML/CMakeLists.txt (hunk 2 above)
Comment (aylward) [reply Wrapping/Generators/CastXML/CMakeLists.txt:40]: When I update to the proper hash, all works!!!

**Comment by thewtex** [comment]
Superceded by #3385

**Pull request closed without merging**